### PR TITLE
System-wide alerts

### PR DESCRIPTION
**Pull request by YanaDePauw**
## References
* Implements #1537
* Requires DSpace/DSpace#8624 

## Description
This PR adds the angular part of the System Wide Alerts feature.

An additional admin menu has been added that will allow administrators to set a system-wide alert.
In the form, a user will need to set a message, optionally set a date for a count down timer and select which sessions will still be supported.

The countdown timer now also supports the setting of a date and time that will allow administrators to put up a maintanance warning with countdown longer in advance than just an hour.
This timer will be displayed in the banner and will be hidden once it expires.
                                                                                 
**Note:** Right now angular does not support any functionality to detect that the session has been disrupted, even when refreshing the token failed, angular will keep the user logged in. 
Due to this issue, restricting authentication is not supported in the system-wide alert as part of this PR. 

## Instructions for Reviewers
To test the system-wide alert perform the following steps: 
* Log in as an administrator and click the new option in the admin sidebar.
* Set an alert message and activate the system-wide alert.
* Verify that a banner shows up at the top of the page.

To test the countdown, add different countdown times and verify that the timer in the banner updates properly.

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [x] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

File: src/app/system-wide-alert/alert-form/system-wide-alert-form.component.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.timepicker-margin {
+    margin-top: -38px;
```
Comment (artlowel) [inline src/app/system-wide-alert/alert-form/system-wide-alert-form.component.scss:2]: Perhaps use a bootstrap `m-` class instead?

File: src/app/system-wide-alert/alert-form/system-wide-alert-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    protected notificationsService: NotificationsService,
+    protected router: Router,
+    protected requestService: RequestService,
+    protected translateService: TranslateService
+  ) {
+  }
+
+  ngOnInit() {
+    this.systemWideAlert$ = this.systemWideAlertDataService.findAll().pipe(
+      getFirstSucceededRemoteDataPayload(),
```
Comment (artlowel) [inline src/app/system-wide-alert/alert-form/system-wide-alert-form.component.ts:66]: What should happen if it fails?

**Review by artlowel — CHANGES_REQUESTED** [review]
Thanks @YanaDePauw!

I think a few things could be clearer:
- While it is explained that the countdown isn't used if it isn't changed, I would make that clearer. E.g. add a toggle in front of the countdown, or perhaps set the countdown section to half opacity until you "activate" it
- It would also be good to show a preview of what that notification will look like. The countdown in that preview doesn't need to work, but e.g. I was surprised to see that it shows up _before_ the message
- I'd also add some more info to the form to make it clear that nothing will happen when the countdown reaches zero. That the admin will still need to turn off the server manually

**Comment by tdonohue** [comment]
@YanaDePauw : Could you rebase this on `main` when you get the chance?  It has merge conflicts, which appear to also be causing the tests to not run (they didn't trigger or startup).  I _think_ a simple rebase/update to latest `main` will fix all that.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@YanaDePauw : I gave this some basic testing today (along with the backend).  The basic functionality works & looks good to me. 

But I ran into a few bugs:
1. First, when the count-down time is enabled in an alert, SSR (server-side rendering) no longer works. No error occurs (that I can locate), but SSR never responds.  Here's how to reproduce:
    * Start frontend in prod mode (`yarn build:prod; yarn serve:ssr`)
    * Enable a SystemWideAlert.  Add a Countdown timer. Set it to "Active".  Click Save.
    * Verify alert appears in your window.  Click around to a different page and verify it appears on all pages (all this works, as it's all CSR)
    * Now, click reload in your browser.  The site will no longer be accessible & will never respond (at least it hasn't for me after 10 minutes)
    * Open a new browser and try to go to your local UI.  It will also not respond (as the first hit is always triggering SSR).
    * Now, go back to the created System Wide Alert, disable the Countdown timer & click Save.  SSR will work again (And now you'll see the alert if you open the site in a new browser & also if you click reload in your browser)
2.  It's possible to select a date/time in the past in the "Countdown timer".  It doesn't cause any error, but I'd expect it to not be possible.  At the very least, the calendar popup should not allow you to select a date prior to today.  Ideally though, you also shouldn't be able to select a time before the current time.
    * As a sidenote, I originally found the clock confusing. I couldn't figure out if the time was in UTC or my local time.  I finally realized it's in local time (which is good).  I'm not sure of any way to make that clearer, but it may be clear enough if you simply cannot select a time earlier than "now".
3. Minor usability issue.  When changing an alert from "Active" to "Inactive" (or visa versa), you are **required** to also click Save to save the changes.  For some reason, that big button looked to me like it should trigger the change immediately.  It took me 2-3 tries in testing this to release that you can only *disable* an active alert by **both** changing it to Inactive and then clicking "Save". 
    * It's possible this is just me being confused...but I wanted to point out that I struggled with this.  I'm not sure how to make this clearer though, so we can leave it as-is if we don't have other ideas & just train people on how to use it.
    
Beyond that, honestly, I like the direction of this feature.  Just needs a few bug fixes.

**Review by tdonohue — COMMENTED** [review]
Thanks @YanaDePauw for the updates!  I retested this today, and my prior feedback is all resolved except for one (very) minor thing:

* It's possible to set a *time* in the past for the Countdown Timer.  Just page backwards an hour or two from the current time.

This honestly is a tiny thing which I'm OK with leaving as-is, unless there's a quick fix.  It doesn't break anything... it's the same behavior as if there was no countdown timer at all. 

Beyond that, I noticed this needs a small rebase (it has minor merge conflicts).  I'm basically a +1 once those merge conflicts are resolved.

(I did also notice that a few of @artlowel 's questions/comments were not addressed yet.. [here](https://github.com/DSpace/dspace-angular/pull/2025#pullrequestreview-1245407543) and [here](https://github.com/DSpace/dspace-angular/pull/2025#discussion_r1068056972).  Nothing seems major though)

**Review by artlowel — APPROVED** [review]
Thanks @YanaDePauw!

**Comment by tdonohue** [comment]
@YanaDePauw : Could you resolve the merge conflicts in this PR?  It looks like it has new minor conflicts..and I cannot move it forward until those are resolved.

**Review by tdonohue — APPROVED** [review]
👍 Thanks @YanaDePauw !  This is now working perfectly for me & all looks good.

**Pull request merged**